index.js

```diff
@@ -12,55 +10,53 @@ function acceptsSerializerPlugin (fastify, options, next) {
 
   const defaultSerializer = globalSerializerManager.findSerializer([options.default])
 
-  fastify.register(require('fastify-accepts'), err => {
-    /* istanbul ignore next */
-    if (err) return next(err)
+  fastify.register(require('fastify-accepts'))
 
-    fastify.addHook('preHandler', (request, reply, done) => {
-      const types = request.types()
-      let serializer
-      let type
+  fastify.addHook('preHandler', (request, reply, done) => {
+    const types = request.types()
+    let serializer
+    let type
 
-      if (!reply.store.config.serializer) {
-        reply.store.config.serializer = {}
-      }
+    if (!reply.serializer) {
+      reply.serializer = {}
+    }
 
-      if (!reply.store.config.serializer.serializerManager) {
-        reply.store.config.serializer.serializerManager = SerializerManager.expand(reply.store.config.serializer, globalSerializerManager)
-      }
+    if (!reply.serializer.serializerManager) {
+      reply.serializer.serializerManager = SerializerManager.expand(reply.serializer, globalSerializerManager)
+    }
 
-      const serializerManager = reply.store.config.serializer.serializerManager
+    const serializerManager = reply.serializer.serializerManager
 
-      const s = serializerManager.findSerializer(types)
-      serializer = s.serializer
-      type = s.type
+    const s = serializerManager.findSerializer(types)
+    serializer = s.serializer
+    type = s.type
 
-      if (!serializer && defaultSerializer) {
-        serializer = defaultSerializer.serializer
-        type = defaultSerializer.type
-      }
+    if (!serializer && defaultSerializer) {
+      serializer = defaultSerializer.serializer
+      type = defaultSerializer.type
+    }
 
-      if (!serializer &&
+    if (!serializer &&
             options.default !== FASTIFY_DEFAULT_SERIALIZE_MIME_TYPE &&
             !request.type(FASTIFY_DEFAULT_SERIALIZE_MIME_TYPE)) {
-        const supportedTypes = serializerManager.getSupportedTypes()
-          .concat([FASTIFY_DEFAULT_SERIALIZE_MIME_TYPE])
-
-        return reply.code(406)
-          .type(FASTIFY_DEFAULT_SERIALIZE_MIME_TYPE)
-          .send(Boom.notAcceptable('Allowed: ' + supportedTypes.join(',')))
-      }
+      const supportedTypes = serializerManager.getSupportedTypes()
+        .concat([FASTIFY_DEFAULT_SERIALIZE_MIME_TYPE])
 
-      if (serializer) {
-        reply.type(type)
-        reply._serializer = serializer.serializeFunction
-      }
+      const notAcceptable = new Error('Allowed: ' + supportedTypes.join(','))
+      return reply.code(406).send(notAcceptable)
+    }
 
-      done()
-    })
+    if (serializer) {
+      reply.type(type)
+      reply._serializer = serializer.serializeFunction
```

Comment:
This work correctly?

Reply:
Strangely yes. You can take this one over if you want.

Reply:
Ok not strangely actually since this is still against fastify 1.x. This work is just for getting everything up to date and marking the transition.